metrics/web/test/features/auth/domain/usercases/sign_in_usecase_test.dart

```diff
@@ -1,15 +1,38 @@
+import 'package:metrics/features/auth/domain/usecases/parameters/user_credentials_param.dart';
 import 'package:metrics/features/auth/domain/usecases/sign_in_usecase.dart';
 import 'package:test/test.dart';
 
 import '../../../../test_utils/matcher_util.dart';
+import 'test_utils/user_repository_stub.dart';
 
 void main() {
-  group("SignInUseeCase", () {
+  group("SignInUseCase", () {
     test("can't be created with null repository", () {
       expect(
         () => SignInUseCase(null),
         MatcherUtil.throwsAssertionError,
       );
     });
+
+    test("signs in user on call", () async {
```

Comment:
It looks like that we are testing stub repository here, and not the use case - I think we just need to see that sign in method was called and that's it. We should probably check 2 cases: successful call - method is executed, 2nd option - throws Authentication Exception - that it's reachable to the caller.

Also I wanted to know how users will know if sig

Reply:
Yes, you are right. You should use the `SingInUseCase` to try to sign in, and if it is successful, the `ReceiveUserUpdates` stream will emit a new user.